cmd/argo-watcher/argo_status_updater.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/rs/zerolog/log"
+	"github.com/shini4i/argo-watcher/internal/helpers"
+	"github.com/shini4i/argo-watcher/internal/models"
+)
+
+type ArgoStatusUpdater struct {
+	argo       Argo
+	retryAttempts uint
+}
+
+func (updater *ArgoStatusUpdater) Init(argo Argo, retryAttempts uint) {
+	updater.argo = argo
+	updater.retryAttempts = retryAttempts
+}
+
+func (updater *ArgoStatusUpdater) WaitForRollout(task models.Task) {
+	// continuously check for application status change
+	status, err := updater.checkWithRetry(task)
+
+	// application synced successfully
+	if status == ArgoAppSuccess {
+		updater.handleDeploymentSuccess(task)
+		return
+	}
+
+	// we had some unexpected error with ArgoCD API
+	if status == ArgoAppFailed {
+		updater.handleArgoAPIFailure(task, err)
+		return
+	}
+
+	// fetch application details
+	app, err := updater.argo.api.GetApplication(task.App)
+
+	// define default message
+	const defaultErrorMessage string = "could not retrieve details"
+	// handle application sync failure
+	switch status {
+	// not all images were deployed to the application
+	case ArgoAppNotAvailable:
+		// show list of missing images
+		var message string
+		// define details
+		if err != nil {
+			message = defaultErrorMessage
+		} else {
+			message = "List of current images (last app check):\n"
```

Comment:
Maybe let's replace multiple "message +=" with a single fmt.Sprintf?

Reply:
I'll try to make it look better